pydicom/valuerep.py

```diff
@@ -654,12 +675,11 @@ def __repr__(self) -> str:
 if config.use_DS_decimal:
     DSclass = DSdecimal
 else:
-    DSclass = DSfloat
+    DSclass = DSfloat  # type: ignore[assignment, misc]
```

Comment:
I'm not sure if this has a legitimate solution apart from refactoring `DSclass` out of existence. The problem is explained [here](https://mypy.readthedocs.io/en/latest/common_issues.html#variables-vs-type-aliases). 

Treating it as an alias (as done here) means we get an assignment error. Treating it as a type variable means it can't be used as a type parameter, which causes errors elsewhere. This seemed the most straightforward.

Reply:
Yes, I've run into similar problems without a solution, I think this is indeed the best way.